lib/tools/android-manifest.js

```diff
@@ -59,6 +62,89 @@ manifestMethods.processFromManifest = async function (localApk) {
  * @property {string} apkActivity - The name of main application activity.
  */
 
+ /**
+ * Extract package and main activity name from application manifest using
+ * the custom apk tools.
+ *
+ * @param {string} localApk - The full path to application package.
+ * @return {APKInfo} The parsed application info.
+ * @throws {Error} If there was an error while getting the data from the given
+ *                 application package.
+ */
+async function extractApkInfoWithApkTools (localApk) {
+  await this.initAapt();
```

Comment:
`this` isn't defined here.

Reply:
thanks, good catch. This, actually failed the tests %)